sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,752 @@
+  az_span time_span = az_span_init((uint8_t*)end_time_buffer, (int32_t)len);
+
+  // Build the method response payload
+  az_json_builder json_builder;
+  AZ_RETURN_IF_FAILED(az_json_builder_init(&json_builder, response, NULL));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_begin_object(&json_builder));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(&json_builder, report_max_temp_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32_number(&json_builder, (int32_t)device_max_temp));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(&json_builder, report_min_temp_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32_number(&json_builder, (int32_t)device_min_temp));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(&json_builder, report_avg_temp_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32_number(&json_builder, (int32_t)device_avg_temp));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(&json_builder, report_start_time_name_span));
+
+  // If the user specified a time, use that as the start | Otherwise use boot time
+  if (az_span_size(start_time_span) > 0)
+  {
+    AZ_RETURN_IF_FAILED(az_json_builder_append_string(&json_builder, start_time_span));
+  }
+  else
+  {
+    AZ_RETURN_IF_FAILED(az_json_builder_append_string(&json_builder, boot_time_span));
+  }
+
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(&json_builder, report_end_time_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_string(&json_builder, time_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_end_object(&json_builder));
+
+  *out_response = az_json_builder_get_json(&json_builder);
+
+  return AZ_OK;
+}
+
+// Send the response of the method invocation
+static int send_method_response(
+    az_iot_hub_client_method_request* request,
+    uint16_t status,
+    az_span response)
+{
+  // Get the response topic to publish the method response
+  if (az_failed(az_iot_hub_client_methods_response_get_publish_topic(
+          &client,
+          request->request_id,
+          status,
+          methods_response_topic,
+          sizeof(methods_response_topic),
+          NULL)))
+  {
+    printf("Unable to get twin document publish topic");
+    return -1;
+  }
+
+  printf("Status: %u\tPayload:", status);
+  char* payload_char = (char*)az_span_ptr(response);
+  if (payload_char != NULL)
+  {
+    for (int32_t i = 0; i < az_span_size(response); i++)
+    {
+      putchar(*(payload_char + i));
+    }
+    putchar('\n');
+  }
+
+  // Send the methods response
+  int rc;
+  if ((rc = MQTTClient_publish(
+           mqtt_client,
+           methods_response_topic,
+           az_span_size(response),
+           az_span_ptr(response),
+           0,
+           0,
+           NULL))
+      != MQTTCLIENT_SUCCESS)
+  {
+    printf("Failed to send method response, return code %d\n", rc);
+    return rc;
```

Comment:
Nit, but this is depending on MQTTCLIENT_SUCCESS being 0.  Based on pushback of having AZ_OK (or whatever it was) be 0, I think we hsould translate this to a -1 or some other code we control.  Similar feedback if we do this other places.

Reply:
This is a topic Cris and I have continued to discuss as well.  I have preferred enums or defined macros since those offer more context, and Cris has pointed out though that because these samples could use other MQTT libraries, we may want to keep all ways to using 0 for a successful return in these MQTT functions.  Ultimately, we need to be consistent across all the samples. TBD?

Reply:
tbd for sure